ui-speedspacechart/src/__tests__/utils.spec.ts

```diff
@@ -270,4 +267,40 @@ describe('getLinearLayerMarginTop', () => {
   });
 });
 
+describe('findPreviousAndNextPosition', () => {
+  it('should return the correct previous and next positions', () => {
+    const xPositionReference = (ref: number) => ref;
+    const cursor = { x: 400, y: 0 };
+    const { previousPosition, nextPosition } = findPreviousAndNextPosition(
+      speeds,
+      cursor.x!,
+      xPositionReference
+    );
+    expect(previousPosition).toEqual({ value: 20, position: { start: 350 } });
+    expect(nextPosition).toEqual({ value: 30, position: { start: 600 } });
+  });
+
+  it('should not return previous position if cursor is at the start of the array', () => {
+    const xPositionReference = (ref: number) => ref;
+    const cursor = { x: 200, y: 0 };
+    const { previousPosition, nextPosition } = findPreviousAndNextPosition(
+      speeds,
+      cursor.x!,
+      xPositionReference
+    );
+    expect(previousPosition).toEqual(nextPosition);
```

Comment:
Don't you think we should test the value of previousPosition too ?
```suggestion
    expect(previousPosition).toEqual(nextPosition);
    expect(previousPosition).toEqual({ value: 20, position: { start: 350 } });
```

Reply:
done